Sources/CardPayments/APIRequests/ConfirmPaymentSourceRequest.swift

```diff
@@ -9,15 +9,21 @@ struct ConfirmPaymentSourceRequest: APIRequest {
     private let orderID: String
     private let pathFormat: String = "/v2/checkout/orders/%@/confirm-payment-source"
     private let accessToken: String
-    private let jsonEncoder = JSONEncoder()
+    var jsonEncoder: JSONEncoder?
     
     /// Creates a request to attach a payment source to a specific order.
     /// In order to use this initializer, the `paymentSource` parameter has to
     /// contain the entire dictionary as it exists underneath the `payment_source` key.
     init(
         accessToken: String,
-        cardRequest: CardRequest
+        cardRequest: CardRequest,
+        encoder: JSONEncoder? = nil
```

Comment:
The other option for this is to create a second init for testing with the JSONEncoder. I think we do this in other files to make it super clear which one is for testing.

Reply:
That makes sense

Reply:
addressed in [commit](4de094c2c79d020a4e248ad4c5677b8d55c02d2e)